src/EntityFramework/Core/Query/PlanCompiler/FilterOpRules.cs

```diff
@@ -144,6 +145,12 @@ private static bool ProcessFilterOverProject(RuleProcessingContext context, Node
                 return false;
             }
 
+            // check to see that this predicate doesn't reference user-defined functions 
+            if (DbProviderServices.DisableFilterSimplificationForCustomFunctions && trc.IncludeCustomFunctionOp(predicateNode, varMap))
```

Comment:
Just in case this is not on purpose: here you are passing the varMap, which was computed for projectNode.Child1 bove. It is really enough or do we need to compute the var map over projectNode?

Reply:
AFAIK, yes, it's enough.
Remapping that happens after new checks:
```
            //
            // Try to remap the predicate in terms of the definitions of the Vars
            //
            var remappedPredicateNode = trc.ReMap(predicateNode, varMap);
``` 
it looks in the predicate and rewrites all references from varMap by real expressions. My check ensures that undesired changes will not happen.

PS. I was thinking about implementing check inside ReMap, but that method ion general purpose and called in a lot of places. I'm not sure about performance impact of doing these checks over and over again.